StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheet+APIMockTest.swift

```diff
@@ -0,0 +1,284 @@
+        stub { urlRequest in
+            guard let pathComponents = urlRequest.url?.pathComponents else { return false }
+            return pathComponents[2] == "payment_intents" && pathComponents.last != "confirm"
+        } response: { request in
+            var json = MockJson.paymentIntent
+
+            // Mock that the PI requires confirmation if it's being fetched for a deferred PI
+            if request.httpMethod == "GET" {
+                json["status"] = "requires_confirmation"
+                json["capture_method"] = "automatic"
+            }
+
+            return HTTPStubsResponse(jsonObject: json, statusCode: 200, headers: nil)
+        }
+
+        stub { urlRequest in
+            guard let pathComponents = urlRequest.url?.pathComponents else { return false }
+            return pathComponents[2] == "setup_intents" && pathComponents.last != "confirm"
+        } response: { request in
+            var json = MockJson.setupIntent
+            // Mock that the PI requires confirmation if it's being fetched for a deferred PI
+            if request.httpMethod == "GET" {
+                json["status"] = "requires_confirmation"
+            }
+
+            return HTTPStubsResponse(jsonObject: json, statusCode: 200, headers: nil)
+        }
+    }
+
+    func testPassthroughModeCallsSharePaymentDetails() {
+        stubConfirmPaymentExpecting(isPaymentIntent: true, paymentMethodId: MockParams.cardPaymentMethod.stripeId)
+        stubLinkShareExpecting(consumerSessionClientSecret: "cs_xxx", paymentMethodID: "pd1")
+        stubLinkLogout(consumerSessionClientSecret: "cs_xxx")
+
+        let configuration = MockParams.configurationWithCustomer(pk: MockParams.publicKey)
+        let exp = expectation(description: "confirm completed")
+        let paymentHandler = STPPaymentHandler(apiClient: configuration.apiClient)
+        let elementsSession = STPElementsSession.linkPassthroughElementsSession
+        PaymentSheet.confirm(
+            configuration: configuration,
+            authenticationContext: self,
+            intent: .deferredIntent(intentConfig: MockParams.deferredPaymentIntentConfiguration(clientSecret: MockParams.paymentIntentClientSecret)),
+            elementsSession: elementsSession,
+            paymentOption: .link(option: .withPaymentDetails(account: .init(email: "test@example.com", session: .init(clientSecret: "cs_xxx", emailAddress: "test@example.com", redactedPhoneNumber: "+1-555-xxx-xxxx", verificationSessions: [.init(type: .sms, state: .verified)], supportedPaymentDetailsTypes: [.card]), publishableKey: MockParams.publicKey), paymentDetails: .init(stripeID: "pd1", details: .card(card: .init(expiryYear: 2055, expiryMonth: 12, brand: "visa", last4: "1234", checks: nil)), billingAddress: nil, billingEmailAddress: nil, isDefault: true))),
+            paymentHandler: paymentHandler,
+            analyticsHelper: ._testValue(),
+            completion: { _, _ in
+                exp.fulfill()
+            }
```

Comment:
Seems like we should do some asserting here? e.g. ensure it has been completed?

Reply:
We do, see the "confirm completed" expectation defined above. Additionally, each of the above stubs have their own expectations.